TCAT/Controllers/FavoritesViewController.swift

```diff
@@ -17,14 +17,37 @@ class FavoritesViewController: UIViewController {
     private let favoritesTitleLabel = UILabel()
     private let tabView = UIView()
     private let tabSize = CGSize(width: 32, height: 4)
+//    public var backgroundDimmingOpacity: CGFloat = 0.5
 
     // MARK: - Data vars
     private let favoritesReuseIdentifier = "FavoritesCollectionViewCell"
-    private var favorites: [String] = ["Collegetown Bagels", "Collegetown Bagels", "Collegetown Bagels", "Collegetown Bagels"] // Temp
+    private let addFavoritesReuseIdentifier = "AddFavoritesCollectionViewCell"
+    private var favoritePlaces: [Place]! {
+        didSet {
+            favoritesCollectionView.reloadData()
+        }
+    }
+    private var isEditingFavorites: Bool! {
+        didSet {
+            setupButtons()
+            favoritesCollectionView.reloadData()
+        }
+    }
+
+    init(isEditing: Bool) {
+        super.init(nibName: nil, bundle: nil)
+        isEditingFavorites = isEditing
+        favoritePlaces = Global.shared.retrievePlaces(for: Constants.UserDefaults.favorites)
```

Comment:
i think if you move `super.init` to the end of the `init`, you can remove the `!` from `favoritePlaces` and `isEditingFavorites`